### PR TITLE
LR Scheduler Adjustments

### PR DESCRIPTION
**Pull request by connor-mccorm**

### CODE REVIEW

File: ludwig/schema/lr_scheduler.py [hunk 1]
```diff
@@ -132,13 +132,7 @@ def _deserialize(self, value, attr, data, **kwargs):
         @staticmethod
         def _jsonschema_type_mapping():
             return {
-                "oneOf": [
```
Comment (ksbrar) [inline ludwig/schema/lr_scheduler.py:135]: So the scheduler will always be enabled?

Comment (connor-mccorm) [reply ludwig/schema/lr_scheduler.py:135]: This is a good point. I don't think we actually want this to always be enabled quite yet - I think if we can figure out a way to set these params automatically then we would. So I actually may revert this part for now.

Comment (connor-mccorm) [reply ludwig/schema/lr_scheduler.py:135]: Ok for the time being, I switched decay to have a default of `None` and be a required param. This is in fact supported so we should be good here. Down the line, I think it's a good idea to think about how to automate the setting of lr scheduler params though.

**Review by tgaddair — APPROVED** [review]

File: ludwig/schema/lr_scheduler.py (hunk 1 above)
Comment (ksbrar) [reply ludwig/schema/lr_scheduler.py:135]: > I switched decay to have a default of None _and be a required param_
What do you mean?

Comment (connor-mccorm) [reply ludwig/schema/lr_scheduler.py:135]: Basically before decay was an optional parameter so we didn't need a default for it. However, decay can take a value of none which just doesn't use any LR decay. So I added that default and removed the `Optional` typing indicator so that `decay` is now a required param from a validation pov but the user doesn't have to specify anything because if they don't it will just default to `None`.
